File: src/main/java/io/openfuture/openmessanger/repository/entity/Attachment.java
package io.openfuture.openmessanger.repository.entity;

import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "attachment")
public class Attachment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String url;

    private Integer messageId;

    private LocalDateTime createdAt;

}

File: src/main/kotlin/io/openfuture/openmessenger/repository/entity/Attachment.kt
package io.openfuture.openmessenger.repository.entity

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "attachment")
class Attachment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
    var name: String? = null
    var url: String? = null
    var messageId: Int? = null
    var createdAt: LocalDateTime? = null
}